models/contacts.js

```diff
@@ -1,14 +1,34 @@
-const fs = require("fs/promises");
-const { nanoid } = require("nanoid");
-const path = require("path");
+const mongoose = require("mongoose");
+const { Schema } = mongoose;
 
-const contactsPath = path.resolve(__dirname, "./contacts.json");
+const contact = new Schema(
+  {
+    name: {
+      type: String,
+      required: [true, "Set name for contact"],
+    },
+    email: {
+      type: String,
+    },
+    phone: {
+      type: String,
+    },
+    favorite: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  {
+    versionKey: false,
+    timestamps: true,
+  }
+);
+
+const Contact = mongoose.model("contact", contact, "contacts");
 
 const listContacts = async () => {
```

Comment:
W poniższych funkcjach raczej nie chcemy używać try/catchy.  Zrobiliśmy obsługę błędu w api/contact.js
`try/catch  -> next(error)` 
I te funkcje są wywoływane z `await` czyli aplikacja czeka za tym aż funkcja się uknończy lub wyrzuci błąd, dlatego tutaj w przypadku błędu nie chcemy z nim nic robić tylko zrzucamy odpowiedzialność za niego do funkcji nadrzędnej

Reply:
Poprawione. Czyli w tych funkcjach, które maja jedynie tworzyć schemat, robić jak najprościej a wyłapywanie błędów robić już w wyższych funkcjach, które już coś powinny zwracać?

Reply:
To zależy jaki rezultat chcesz osiągnąć. Jeśli miał byś ochotę użyć tam try/catch'y bo np  masz wiele rzeczy które mogą wyrzucić błąd i dla każdego masz inny customowy error to możesz. Ale w tym wypadku tu i tak zwrócisz wszędzie 500 bo jest coś nie tak z serverem i jeśli tu wywali błąd to await go przekaże wyżej. To może trochę być skomplikowane więc doradzam przeciwiczyć czy poszukać jakiś przykładów.